lib/bunny/session.rb

```diff
@@ -797,12 +801,16 @@ def recover_channels
       @channel_mutex.synchronize do
         @channels.each do |n, ch|
           ch.open
-
           ch.recover_from_network_failure
         end
       end
     end
 
+    # @private
+    def network_recovery_callback
+      @network_recovery_callback&.call
```

Comment:
Not sure what the trailing `&` does in this case but we can leave it with just `.call` as we do with all similar methods that accept callables.

Reply:
hi, by default the callback will be `nil`, `&.` is the safe navigation operator to prevent nil class errors

Reply:
I have been using Ruby since 2005 and never knew about it. Can we please use a more obvious `… if callback` guard?